sdk/CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+# SDK Contribution Guide
+
+This document serves as a guide for creating a new Dagger SDK, but also to help maintain consistency between existing ones.
+
+## Purpose of an SDK
+
+- Best DX
```

Comment:
I'd say "best" is subjective, could we expand on this with specifics?

Reply:
Of course, this whole first draft is just a brainstorm to be expanded later.